docs/source/delta-batch.md

```diff
@@ -407,6 +407,150 @@ You can use an [EXPLAIN](https://spark.apache.org/docs/latest/sql-ref-syntax-qry
+    .addColumn("id_col3", dataType=LongType(), generatedByDefaultAs=IdentityGenerator())
+    .addColumn("id_col4", dataType=LongType(), generatedByDefaultAs=IdentityGenerator(start=-1, step=1))
+    .execute()
+  ```
+
+  ```scala
+  import io.delta.tables.DeltaTable
+  import org.apache.spark.sql.types.LongType
+
+  DeltaTable.create(spark)
+    .tableName("table_name")
+    .addColumn(
+      DeltaTable.columnBuilder(spark, "id_col1")
+        .dataType(LongType)
+        .generatedAlwaysAsIdentity().build())
+    .addColumn(
+      DeltaTable.columnBuilder(spark, "id_col2")
+        .dataType(LongType)
+        .generatedAlwaysAsIdentity(start = -1L, step = 1L).build())
+    .addColumn(
+      DeltaTable.columnBuilder(spark, "id_col3")
+        .dataType(LongType)
+        .generatedByDefaultAsIdentity().build())
+    .addColumn(
+      DeltaTable.columnBuilder(spark, "id_col4")
+        .dataType(LongType)
+        .generatedByDefaultAsIdentity(start = -1L, step = 1L).build())
+    .execute()
+  ```
+
+.. note:: SQL APIs for identity columns are not supported yet.
+
+You can optionally specify the following:
+- A starting value.
+- A step size, which can be positive or negative.
+
+Both the starting value and step size default to `1`. You cannot specify a step size of `0`.
+
+Values assigned by identity columns are unique and increment in the direction of the specified step, and in multiples of the specified step size, but are not guaranteed to be contiguous. For example, with a starting value of `0` and a step size of `2`, all values are positive even numbers but some even numbers might be skipped.
+
+When the identity column is specified to be `generated by default as identity`, insert operations can specify values for the identity column. Specify it to be `generated always as identity` to override the ability to manually set values.
+
+Identity columns only support `LongType`, and operations fail if the assigned value exceeds the range supported by `LongType`.
+
+You can use `ALTER TABLE table_name ALTER COLUMN column_name SYNC IDENTITY` to synchronize the metadata of an identity column with the actual data. When you write your own values to an identity column, it might not comply with the metadata. This option evaluates the state and updates the metadata to be consistent with the actual data. After this command, the next automatically assigned identity value will start from `start + (n + 1) * step`, where `n` is the smallest value that satisfies `start + n * step >= max()` (for a positive step).
+
+#### CTAS and identity columns
+
+You cannot define schema, identity column constraints, or any other table specifications when using a `CREATE TABLE table_name AS SELECT` (CTAS) statement.
+
+To create a new table with an identity column and populate it with existing data, do the following:
+
+#. Create a table with the correct schema, including the identity column definition and other table properties.
+#. Run an insertion operation.
```

Comment:
What does `#.` look like? have you built this page and looked at the html?

Reply:
<img width="884" alt="Screenshot 2024-12-11 at 21 57 15" src="https://github.com/user-attachments/assets/0c49d138-4e20-4879-be17-9246c5df772a" />